assets/src/edit-story/app/story/actions/useAutoSave.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import objectPick from '../../../utils/objectPick';
+import { useAPI } from '../../api';
+import { useConfig } from '../../config';
+import getStoryMarkup from '../../../utils/getStoryMarkup';
+
+/**
+ * Custom hook to auto-save a story.
+ *
+ * @param {Object}    properties Properties to update.
+ * @param {number}    properties.storyId Story post id.
+ * @param {Array}     properties.pages Array of all pages.
+ * @param {Object}    properties.story Story-global properties
+ * @return {Function} Function that can be called to save a story.
+ */
+function useAutoSave({ storyId, pages, story }) {
+  const {
+    actions: { autoSaveById },
+  } = useAPI();
+  const { metadata } = useConfig();
+  const [isAutoSaving, setIsAutoSaving] = useState(false);
+
+  const autoSave = useCallback(
+    (props) => {
+      setIsAutoSaving(true);
+      const propsToSave = objectPick(story, [
+        'title',
+        'status',
+        'author',
+        'date',
+        'modified',
+        'slug',
+        'excerpt',
+        'featuredMedia',
+        'password',
+        'publisherLogo',
+        'stylePresets',
+        'autoAdvance',
+        'defaultPageDuration',
+      ]);
+      const content = getStoryMarkup(story, pages, metadata);
+      return autoSaveById({
+        storyId,
+        content,
+        pages,
+        ...propsToSave,
+        ...props,
+      }).finally(() => setIsAutoSaving(false));
+    },
+    [story, pages, metadata, autoSaveById, storyId]
+  );
+
+  return { autoSave, isAutoSaving };
```

Comment:
Is `isAutoSaving` the same as `isSaving` from `useSaveStory`? And overall, does this overlap at all with `useSaveStory`? Should it? E.g. should this function call `useSaveStory`? Otherwise, this code looks mostly repeated.

Reply:
I'm not sure, thought about it at first, however, there's the difference that auto-save isn't actually saving the story, it felt more clear to separate it from `useSaveStory`. There might also be more auto-save creation-related nuances that come with #221 
`isAutoSaving` is the same as `isSaving` in terms of both reflecting the API request not having completed, it's a different request to a different endpoint though, not sure if these two should manipulate the same `isSaving`, especially since auto-saving will be triggered every 60 seconds automatically.

Thoughts?

Reply:
Thinking about it more, then when auto-saving is implemented as automated every 60 seconds, then it does have a different meaning from `isSaving` -- the user shouldn't really notice that auto-saving is happening in the background.

Reply:
Thanks @miina! These are good reasons to keep separate then.

Reply:
That being said, it still looks like a lot of content here is duplicated vs `useSaveStory`: `objectPick`, `getStoryMarkup`. Is there still a way to share more while keeping them separate?

Reply:
Added an util for sharing more code: e690cf5